src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/eventDetailsBuilder.hpp

```diff
@@ -229,6 +229,9 @@ class TEventDetailsBuilder final : public AbstractHandler<std::shared_ptr<TScanC
                 auto vulnerabilityDetection = PolicyManager::instance().getVulnerabilityDetection();
                 ecsData["wazuh"]["cluster"]["name"] = data->clusterName();
                 ecsData["wazuh"]["schema"]["version"] = WAZUH_SCHEMA_VERSION;
+                ecsData["wazuh"]["vulnerability"]["under_evaluation"] =
+                    !(Utils::floatToDoubleRound(returnData.data->scoreBase(), 2) &&
+                      !returnData.data->classification()->str().empty() && !returnData.data->severity()->str().empty());
```

Comment:
Improve (complexity): I would move this to a lambda or a private method to reduce cognitive complexity

```c++
                const auto isUnderEvaluation = [&returnData]()
                {
                    if (Utils::floatToDoubleRound(returnData.data->scoreBase(), 2) == 0)
                    {
                        return false;
                    }

                    if (returnData.data->classification()->str().empty())
                    {
                        return false;
                    }

                    if (returnData.data->severity()->str().empty())
                    {
                        return false;
                    }

                    return true;
                };

                ecsData["wazuh"]["vulnerability"]["under_evaluation"] = isUnderEvaluation();
                ```

Reply:
Thank you! Nice catch! Just a quick note on why we refactored the `isUnderEvaluation` function in our last commit in a different way

We streamlined the logic into a single return statement, making it much easier to read and maintain. This also cuts down on repetitive `if` checks and improves efficiency with short-circuit evaluation. On top of that, it’ll be simpler to update if we need to add more conditions in the future.

Reply:
I liked the mix between both ideas!